### PR TITLE
RavenDB-12365: Replace unmanaged memory pInvokes with Unsafe calls

### PR DESCRIPTION
**Pull request by aviviadi**

### CODE REVIEW

**Comment by ppekrol** [comment]
@aviviadi Can you fix compilation?

File: src/Sparrow/Platform/Win32/Win32ElectricFencedMemory.cs [hunk 1]
```diff
@@ -31,7 +31,7 @@ internal unsafe class Win32ElectricFencedMemory
 
             var firstWritablePage = virtualAlloc + 4096;
 
-            Win32UnmanagedMemory.Set(firstWritablePage, 0xED, 4096 * sizeInPages); // don't assume zero'ed mem
```
Comment (arekpalinski) [inline src/Sparrow/Platform/Win32/Win32ElectricFencedMemory.cs:34]: Can we remove `Win32UnmanagedMemory.cs` file now?

Comment (aviviadi) [reply src/Sparrow/Platform/Win32/Win32ElectricFencedMemory.cs:34]: No, we still use : memcmp, memove, which doesn't have Unsafe methods and we probably do not want to implement them by ourselfs (or may be we will..?)

**Review by ayende — APPROVED** [review]

File: src/Sparrow/Platform/Win32/Win32ElectricFencedMemory.cs (hunk 1 above)
Comment (arekpalinski) [reply src/Sparrow/Platform/Win32/Win32ElectricFencedMemory.cs:34]: ok, no need to implement them on our own, we're better of by using those syscals I think

Comment (aviviadi) [reply src/Sparrow/Platform/Win32/Win32ElectricFencedMemory.cs:34]: in 4.2 we replace memove with Buffer.MemoryCopy

**Pull request merged**